### PR TITLE
Adds a new pneumatic shotgun

### PR DESCRIPTION
**Pull request by ddsaw**
#### SUMMARY
SUMMARY: Content "Adds a new double barrel pneumatic shotgun"

#### Purpose of change
Currently there is no good shotgun to use for training, and no "low power silent, available ammo" option, either. Since shotgun silencers are very rare, I felt like a pneumatic shotgun like this could do the trick.

#### Describe the solution
The implemented gun is quite a bit weaker compared to a standard double barreled shotgun, both in reload time and damage. Ammo comes in a few varieties with the best consuming 10 flechettes per craft and the worst simple taking pebbles and scrap off the ground.
Currently there is no crafting recipe for the gun itself because I didn't find the recipe for the other pneumatics guns in the recipes: ranged json, but it will be equivalent to what the other pneumatics take.

#### Describe alternatives you've considered
I've considered making it single barrel aswell as reducing it's damage but after some testing comparing it against the blunderbuss and double barreled I believe I should get some other opinions first, considering how controversial pneumatics apparently are.
 I would consider adding a recipe book like "Tactical Airgun Digest" or something as a crafting requirement to all pneumatics, but that would better be handled in a separate issue (also I would have to find these item recipes first).

### CODE REVIEW

**Comment by LaVeyanFiend** [comment]
Do you have a real-life version to compare this gun to?

**Comment by ddsaw** [comment]
> Do you have a real-life version to compare this gun & ammo to?
> 
> (I really want this since every other ranged weapon skill has a cheap way to train them, including launchers via spraycan flamethrower, but I don't want it to be "just because I want it"  )

I'm thinking about the many subsistence hunting airshotguns that are used in countries where conventional shotguns are banned.  Examples are the Falco Arms Air shotguns aswell as the korean Yewha shotguns. As for being double barreled look into the Seneca Doubleshot.
The Yewha specifically is a handpumped, hand crafted(or so I think), cheap airshotgun albeit single barrel, though a double barreled design wouldn't be unfeasible to make.
The ammo is inspired by those same guns(they're functionally shotgunshells without powder), but I will gladly change it to require plastic molds or whatever aswell as a few plastic chunks for the hull if you think a paper shell wouldn't be enough.

File: data/json/items/ammo/shotcanister.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
+[
+  {
+    "type": "AMMO",
+    "id": "shotcanister_scrap",
+    "price": 1000,
+    "name": "scrap shotcanister",
+    "symbol": "=",
+    "color": "green",
+    "description": "This small paper canister resembling a crude shotshell is filled to the brim with assorted pieces of scrap metal.",
+    "material": ["steel" ],
```
Comment (Night-Pryanik) [inline data/json/items/ammo/shotcanister.json]:
```suggestion
    "material": [ "steel" ],
```

File: data/json/items/ammo/shotcanister.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  },
+  {
+    "type": "AMMO",
+    "id": "shotcanister_pebble",
+    "price": 1000,
+    "name": "pebble shotcanister",
+    "symbol": "=",
+    "color": "green",
+    "description": "This small paper canister resembling a crude shotshell is filled to the brim with tiny pebbles. Damage is pathethic but flight is much more stable than scrap",
+    "material": ["steel" ],
```
Comment (Night-Pryanik) [inline data/json/items/ammo/shotcanister.json]:
```suggestion
    "material": [ "steel" ],
```

File: data/json/items/ammo/shotcanister.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    "count": 10
+  },
+  {
+    "type": "AMMO",
+    "id": "shotcanister_pebble",
+    "price": 1000,
+    "name": "pebble shotcanister",
+    "symbol": "=",
+    "color": "green",
+    "description": "This small paper canister resembling a crude shotshell is filled to the brim with tiny pebbles. Damage is pathethic but flight is much more stable than scrap",
```
Comment (Night-Pryanik) [inline data/json/items/ammo/shotcanister.json]:
```suggestion
    "description": "This small paper canister resembling a crude shotshell is filled to the brim with tiny pebbles.  Damage is pathetic but flight is much more stable than scrap",
```

File: data/json/items/ammo/shotcanister.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    "material": ["steel" ],
+    "volume": 1,
+    "weight": 55,
+    "bashing": 2,
+    "cutting": 1,
+    "ammo_type": "shotcanister",
+    "looks_like": "shot_scrap",
+    "range": 2,
+    "pierce" : 1,
```
Comment (Night-Pryanik) [inline data/json/items/ammo/shotcanister.json]:
```suggestion
    "pierce": 1,
```

File: data/json/items/ammo/shotcanister.json [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+
+  {
+    "type": "AMMO",
+    "id": "shotcanister_flechette",
+    "price": 1000,
+    "name": "flechette shotcanister",
+    "symbol": "=",
+    "color": "green",
+    "description": "This small paper canister resembling a crude shotshell is filled to the brim with handmade flechettes, giving it some armor piercing capabilities. ",
```
Comment (Night-Pryanik) [inline data/json/items/ammo/shotcanister.json]:
```suggestion
    "description": "This small paper canister resembling a crude shotshell is filled to the brim with handmade flechettes, giving it some armor piercing capabilities.",
```

File: data/json/items/ammo/shotcanister.json [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    "material": ["steel" ],
```
Comment (Night-Pryanik) [inline data/json/items/ammo/shotcanister.json]:
```suggestion
    "material": [ "steel" ],
```

File: data/json/items/ammo/shotcanister.json [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  },
+
+  {
+    "type": "AMMO",
+    "id": "shotcanister_bearing",
+    "price": 1000,
+    "name": "bearing shotcanister",
+    "symbol": "=",
+    "color": "green",
+    "description": "This small paper canister resembling a crude shotshell is filled to the brim with handmade bearings, stabilizing it's flight pattern a tiny bit. ",
```
Comment (Night-Pryanik) [inline data/json/items/ammo/shotcanister.json]:
```suggestion
    "description": "This small paper canister resembling a crude shotshell is filled to the brim with handmade bearings, stabilizing it's flight pattern a tiny bit.",
```

File: data/json/items/ranged.json [hunk 8]
```diff
@@ -318,6 +318,34 @@
...
   },
+  {
+    "id": "pneumatic_shotgun",
+    "type": "GUN",
+    "reload_noise_volume": 10,
+    "symbol": "(",
+    "color": "dark_gray",
+    "looks_like": "pipe_double_shotgun",
+    "name": "pneumatic shotgun",
+    "description": "A double-barreled pneumatic air shotgun handcrafted from scrap. Though it's firepower is lacking compared to more conventional shotguns, this thing can still pack quite a punch. That is, if your target is directly infront of you.",
```
Comment (Night-Pryanik) [inline data/json/items/ranged.json]:
```suggestion
    "description": "A double-barreled pneumatic air shotgun handcrafted from scrap.  Though it's firepower is lacking compared to more conventional shotguns, this thing can still pack quite a punch.  That is, if your target is directly in front of you.",
```

File: data/json/recipes/recipe_ammo.json [hunk 9]
```diff
@@ -619,6 +619,78 @@
...
+    "category": "CC_AMMO",
+    "subcategory": "CSC_AMMO_SHOT",
+    "skill_used": "fabrication",
+    "skills_required": [ "gun", 1 ],
+    "difficulty": 3,
+    "time": "1 m",
+    "autolearn": true,
+    "book_learn": [ [ "recipe_bullets", 2 ], [ "manual_shotgun", 2 ] ],
+    "qualities": [],
+    "using": [ ],
```
Comment (Night-Pryanik) [inline data/json/recipes/recipe_ammo.json]: Both of these empty fields can be omitted.

**Comment by Night-Pryanik** [comment]
Please run your changes through the [json linter](http://dev.narc.ro/cataclysm/format.html).

**Comment by ddsaw** [comment]
> Please run your changes through the [json linter](http://dev.narc.ro/cataclysm/format.html).

Yeah sorry, I didn't really understand how to use it or what it does, but I think I get it now. It basically removes stupid formatting errors right? Do I just have to copypaste literally every file I touch and run it through there and then paste it back in?

**Comment by Night-Pryanik** [comment]
> It basically removes stupid formatting errors right?

Well, yeah.

> Do I just have to copypaste literally every file I touch and run it through there and then paste it back in?

Well, yeah.

**Comment by Rivet-the-Zombie** [comment]
The ammo could really use a carrier, perhaps the extant paper cartridge pouch?

**Comment by ddsaw** [comment]
> The ammo could really use a carrier, perhaps the extant paper cartridge pouch or shotgun bandoliers?

I didn't understand how to add a second ammotype to the bandolier, and my attempts at doing so failed. I considered adding it's own bandolier and infact I got that working, but in my head I couldn't justifiy not using the extant versions.

**Pull request merged**